test/distributed/org/apache/cassandra/distributed/mock/nodetool/InternalNodeProbe.java

```diff
@@ -45,14 +47,30 @@
 import org.apache.cassandra.service.GCInspector;
 import org.apache.cassandra.service.StorageProxy;
 import org.apache.cassandra.service.StorageService;
+import org.apache.cassandra.service.StorageServiceMBean;
 import org.apache.cassandra.streaming.StreamManager;
 import org.apache.cassandra.tools.NodeProbe;
+import org.mockito.Mockito;
 
 public class InternalNodeProbe extends NodeProbe
 {
-    public InternalNodeProbe() throws IOException
+    private final boolean withNotifications;
+
+    public InternalNodeProbe(boolean withNotifications) throws IOException
     {
         super("", 0);
+        this.withNotifications = withNotifications;
+        connect(); // need to double "connect" since constructor calls connect; withNotifications isn't visable the first time
```

Comment:
how about adding a new `NodeProble` constructor with param `boolean connect`, so we don't have to connect twice...

Reply:
done